clusterous/cluster.py

```diff
@@ -735,33 +933,52 @@ def init_cluster(self, cluster_name, cluster_spec, nodes_info=[], logging_level=
-                res = conn.run_instances(defaults.node_ami_id, min_count=num_nodes, max_count=num_nodes,
-                                    key_name=c['key_pair'], instance_type=instance_type,
-                                    subnet_id=c['subnet_id'], block_device_map=node_block_devices, security_group_ids=[sg_id])
+    
+                res = conn.run_instances(defaults.node_ami_id, 
+                                         min_count=num_nodes, 
+                                         max_count=num_nodes,
+                                         key_name=c['key_pair'], 
+                                         instance_type=instance_type,
+                                         block_device_map=node_block_devices, 
+                                         subnet_id=private_subnet.id, 
+                                         security_group_ids=[private_security_group.id])
                 node_tags = {'Name': defaults.node_name_format.format(cluster_name, node_tag),
                             defaults.instance_node_type_tag_key: node_tag}
                 node_tags_and_res.append((node_tag, node_tags, res.instances))
-
+    
             # Launch logging instance if necessary
             logging_tags_and_res = []
             if logging_level > 0:
                 self._logger.info('Starting central logging instance')
-                logging_res = conn.run_instances(defaults.central_logging_ami_id, min_count=1,
-                                                key_name=c['key_pair'], instance_type=defaults.central_logging_instance_type,
-                                                subnet_id=c['subnet_id'], block_device_map=block_devices, security_group_ids=[sg_id])
+    
+                logging_res = conn.run_instances(defaults.central_logging_ami_id, 
+                                                 min_count=1,
+                                                 key_name=c['key_pair'], 
+                                                 instance_type=defaults.central_logging_instance_type,
+                                                 block_device_map=block_devices, 
+                                                 subnet_id=private_subnet.id, 
+                                                 security_group_ids=[private_security_group.id])
                 logging_tags = {'Name': defaults.central_logging_name_format.format(cluster_name),
                                 defaults.instance_node_type_tag_key: defaults.central_logging_name_tag_value}
                 logging_tags_and_res = [('central-logging', logging_tags, logging_res.instances)]
-
-
+    
+    
             # Wait for controller to launch
             controller = self._wait_and_tag_instance_reservations(controller_tags_and_res)
-
+            if not retry_ssh(nat_instance.ip_address, defaults.nat_ssh_port_forwarding, 5, 45):
+                raise ClusterException('Error: Unable to SSH Controller instance.')
```

Comment:
For consistency, it's best not to have the "Error:" prefix in the exception messages. This is because the exception handler code is the best place to provide context, e.g. "There was an error launching the cluster: [exception message]".